internal/manifests/collector/container.go

```diff
@@ -104,6 +104,15 @@ func Container(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelem
 		},
 	})
 
+	if len(otelcol.Spec.ConfigMaps) > 0 {
+		for keyCfgMap := range otelcol.Spec.ConfigMaps {
+			volumeMounts = append(volumeMounts, corev1.VolumeMount{
+				Name:      naming.ConfigMapExtra() + otelcol.Spec.ConfigMaps[keyCfgMap].Name,
+				MountPath: otelcol.Spec.ConfigMaps[keyCfgMap].MountPath + "/" + otelcol.Spec.ConfigMaps[keyCfgMap].Name,
```

Comment:
What if MountPath is just `/`? That sounds valid and would break in this case. We should probably use https://pkg.go.dev/path#Join here.

Reply:
Thank you @swiatekm-sumo for pointing out all these topics. I've just changed the PR accordingly. Could you please have a look?